### ISSUE TITLE
Accessibility: Important dialog missing role and out of meaningful reading order

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52144]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The purpose of the store notice is to provide important information to website visitors. Shop owners may use it for cookie notices, sale information, alerts about shipping interruptions and other timely or important information.

Currently, this banner is located after all other content in the DOM and not meaningfully named or targeted. Screen reader users are likely to miss the store notice and not even know it is there.

![Image](https://github.com/user-attachments/assets/4a898ce9-e301-4b47-b6f5-94b9cbc2d361)

### Component or Page: 

WooCommerce Store Notice

### Name affected element:

Container for Store Notice

### Relevant code snippets:

`<p class="woocommerce-store-notice demo_store" data-notice-id="fe68900dee1653cf2d0662bb68d31594" style="display: block;">This is a demo store for testing purposes — no orders shall be fulfilled. <a href="#" class="woocommerce-store-notice__dismiss-link">Dismiss</a></p>`

### Recommended fix:

The store notice should be treated as a dialog. Dialogs can be either non-modal (it's still possible to interact with content outside of the dialog) or modal (only the content in the dialog can be interacted with).

First, you need to decide if users should be forced to interact with the dialog (by dismissing it) before they can interact with the page content or if they should be able to interact with page content without dismissing it and then adjust the code accordingly. Our recommendation is to treat it as a modal and force interaction.

Required fixes:
1. Add role=""dialog"" to the wrapper container.
2. Add aria-modal=""true"" on the wrapper container (if treating as a modal).
3. When the store notice appears, set keyboard focus one the store notice wrapper so screen readers read the content.
4. When the Dismiss button is clicked, return keyboard focus to the Skip to Content link at the top of the page.
5. Add support for dismissing the store notice with the escape key.
6. (If not treating as a modal), Tabbing past the Dismiss button should take users to the Skip to Content link.  

You may want to move this from the bottom of the DOM to before the skip link to improve the tab order if not treating it as a modal.
 
Learn more about the dialog role: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role
